test/skate_web/controllers/page_controller_test.exs

```diff
@@ -52,6 +52,29 @@ defmodule SkateWeb.PageControllerTest do
       assert html =~ "selected_route_ids&quot;:[&quot;1&quot;]"
     end
 
+    @tag :authenticated
+    test "includes user test groups in HTML", %{conn: conn, user: user} do
+      user_struct = user |> Skate.Settings.User.get()
+
+      Skate.Settings.TestGroup.update(%{
+        Skate.Settings.TestGroup.create("html-test-group")
+        | users: [user_struct]
+      })
+
+      user_struct = user |> Skate.Settings.User.get() |> Skate.Repo.preload(:test_groups)
+
+      conn = get(conn, "/")
+
+      html = html_response(conn, 200)
+
+      json =
+        Jason.encode!(user_struct.test_groups |> Enum.map(& &1.name))
+        |> Phoenix.HTML.html_escape()
+        |> Phoenix.HTML.safe_to_string()
```

Comment:
Does this read okay? would it read more "idomatically" if instead it was like this?
```suggestion
      json =
      	user_struct.test_groups
      	|> Enum.map(& &1.name)
        |> Jason.encode!()
        |> Phoenix.HTML.html_escape()
        |> Phoenix.HTML.safe_to_string()
```

I like it being within the `encode` parentheses because it makes it clear what's happening to the `test_groups` first but maybe there's another way to format it and reduce cognitive overhead via indentation/shape?

Reply:
I personally tend to find it easier to read pipe chains when they start with a single value, like your suggested change does. The first thing that is happening to the test groups is that they are being mapped to only their name, so for me personally it is easier to see that as a distinct step. 

Much of my elixir preferences were informed by working on a project with "controversial" credo checks enabled, for example, the credo check for [pipe_chain_start](https://github.com/rrrene/credo/blob/master/lib/credo/check/refactor/pipe_chain_start.ex) which applies in this scenario. So I acknowledge that this may be a "controversial" preference, and that particular credo check is not enabled in this repo, so I wouldn't block a PR over it. Over all though, I found credo checks to be a helpful guide for picking up conventions as I was starting to learn elixir, and have adopted or rejected some of the "controversial" ones over time.

Reply:
I also prefer starting the pipe with a single value.